src/lib.rs

```diff
@@ -673,18 +776,215 @@ decl_module! {
+             */
+
+            // Get category
+            let category = category_tree_path.first().unwrap().clone();
+
+            assert_eq!(category.id, category_id);
+            
+            // Update (unmoderated) thread count in corresponding category
+            <CategoryById<T>>::mutate(category_id, |c| {
+                c.num_direct_unmoderated_threads += 1;
+            });
+
+            // Get thread counter
+            let new_thread_id = <NextThreadId<T>>::get();
+
+            // Create and add new thread
+            let new_thread = Thread {
+                id : new_thread_id,
+                title : title.clone(),
+                category_id: category_id,
+                thread_nr_in_category: category.num_threads_ever_created(),
+                moderation : None,
+                num_unmoderated_posts: 0,
+                num_moderated_posts: 1,
+                created_at : Self::current_block_and_time(),
+                author_id : who.clone()
+            };
+
+            <ThreadById<T>>::insert(new_thread_id, new_thread);
+
+            // Update next thread id
+            <NextThreadId<T>>::mutate(|n| {
+                *n += 1;
+            });
+
+            // Make and add initial post
+
+            // TODO: perhaps factor out later?
+
+            let new_post_id = <NextPostId<T>>::get();
+
+            <NextPostId<T>>::put(new_post_id + 1);
+
+            // Create and add new post
+            let new_post = Post {
+                id: new_post_id,
+                thread_id: new_thread_id,
+                post_nr_in_thread: 1, // Starts at 1
+                current_text: text.clone(),
+                moderation : None,
+                expired_post_texts: vec![],
+                created_at : Self::current_block_and_time(),
+                author_id : who.clone()
+            };
+
+            <PostById<T>>::insert(new_post_id, new_post);
+
+            // Generate event
+            Self::deposit_event(RawEvent::ThreadCreated(new_thread_id));
+
+            Ok(())
+        }
+
+        /// Moderate thread
+        fn moderate_thread(origin, thread_id: ThreadId, rationale: Vec<u8>) -> dispatch::Result {
+
+            // Check that its a valid signature
+            let who = ensure_signed(origin)?;
+            
+            // Signed by forum SUDO
+            Self::ensure_is_forum_sudo(&who)?;
+
+            // Get thread
+            let mut thread = Self::ensure_thread_exists(&thread_id)?;
+
+            // Thread is not already moderated
+            ensure!(thread.moderation.is_none(), ERROR_THREAD_ALREADY_MODERATED);
+
+            // Rationale valid
+            ensure_thread_moderation_rationale_is_valid(&rationale)?;
+
+            // Can mutate in corresponding category
+            let path = Self::build_category_tree_path(thread.category_id);
+
+            // Path must be non-empty, as category id is from thread in state
+            assert!(path.len() > 0);
```

Comment:
```suggestion
            assert!(path.is_empty());
```

Reply:
Changed to `!path.is_empty()`, as its asserting non-emptyness.